src/rgw/rgw_common.h

```diff
@@ -1297,6 +1297,7 @@ struct req_state {
   /* aws4 auth support */
   bool aws4_auth_needs_complete;
   bool aws4_auth_streaming_mode;
+  bool aws4_auth_presigned_url_bugfix;
```

Comment:
'rgw_s3_auth_aws4_presigned_url_bugfix' can be used to decide whether to compute the signature once or twice, eliminating the need for this variable - 'aws4_auth_presigned_url_bugfix.